uniconnect-epfl/uniconnect
# feat: added the events a person is attending/has attended in the database

# What I did
- I created a new field in the User type for storing his events
- When a user creates a new Event, it is added on his list of events, and then displayed on his personal profile (display not done in this PR, but the eventId is passed to the proper screen)
- When a user registers to an existing event, it is also added to the list of events
- Added a function to retrieve an event with his event Id, and connected it to the `ViewEventScreen`

# How I did it

As it is a big PR, I had to change multiple things in multiple files to cover the whole event flow:
- I added the `uid` as a field in the `event`, `announcement` and in the `user` documents
- I created a function in `ManageEvents` that given a `eventId` retrieves all its information
- I created a field in the `user` document that keeps track of the events where the user has registered, and the ones that he has created
- I linked the above function to `EventCreationScreen`
- In the `ProfileEvents` file, I created a flow that gets the actual user and fetches all of his `eventId` 
- I changed the `ViewEventScreen.tsx` so that the right information is shown about an event
 
# How to verify it

As in the video, you can try to create an event and see if the `eventId` appears in your profile screen. You can also try to click on the button participate. 
To test that you indeed do not register twice, you can try to register twice to an event.

# Demo video

https://github.com/uniconnect-epfl/uniconnect/assets/91184289/d74dd3be-067f-41a7-a50c-f386a6518d7e

![Capture d’écran 2024-05-14 à 11 38 20](https://github.com/uniconnect-epfl/uniconnect/assets/91184289/e077ddfd-4f58-4afe-ac84-2bda490394d7)
![Capture d’écran 2024-05-14 à 11 38 32](https://github.com/uniconnect-epfl/uniconnect/assets/91184289/97430b37-f4fc-4620-9e27-26168f724678)




# Pre-merge checklist
The changes I have introduced:
